bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/SerialClient.java

```diff
@@ -69,7 +71,8 @@ protected KNXNetworkLink establishConnection() throws KNXException, InterruptedE
                     e);
         } catch (KNXException e) {
             final String msg = e.getMessage();
-            if ((msg != null) && (msg.startsWith("can not open serial port"))) {
+            // TODO add a test for this string match; error message might change in later version of Calimero library
+            if ((msg != null) && (msg.startsWith(CALIMERO_ERROR_CANNOT_OPEN_PORT))) {
```

Comment:
Just curious: Can you point to the code throwing this exception? What is the actual exception - raw `KNXException` or a subclass? Does it have cause within, e.g. `IOException`? If there is really not enough information to handle it appropriately besides using the message, perhaps it would be worth filing an issue (or pull request) towards the Calimero repository?

Reply:
@jlaur  I agree, this is not the perfect way to do it (but adapting the text is a quick solution for 3.3 which at least makes the feature work again).
https://github.com/calimero-project/calimero-core/blob/0442d65f39ce0a2c39ce8d4674465de62bee6d11/src/tuwien/auto/calimero/serial/LibraryAdapter.java#L168
Maybe it is; the handling of serial devices in the KNX needs to be changed anyway and due to usage of gnu.io library (#7573) and because there have been changes on the Calimero main branch in the area serial communication.

Reply:
If the exception is thrown by that line, it includes the original exception, which can be obtained by `e.getCause()`. Perhaps the original exception provides more useful information?

Reply:
Yes, good point. But I am still not convinced that this helps much in this case, as it is depending on underlying com stack implementation. For me it is ```gnu.io.NoSuchPortException```, but it could also be ```gnu.io.PortInUseException``` or ```gnu.io.UnsupportedCommOperationException```.
Borderline is: port cannot be opened. This is what the initial implementation tried to detect via the string comparison. It was broken by the update to Calimero 2.5 as the text changed.

Reply:
OK, fair enough. Fix is fine for now. If we want a more robust solution, an issue for throwing a dedicated exception (subclass of `KNXException`) could be created for the Calimero project.